File: ios/OTPPoc/OTPPoc/Models/AuthModels.swift
import Foundation

struct ApiResponse<T: Decodable>: Decodable {
    let success: Bool
    let data: T?
    let error: ApiErrorBody?
}

struct ApiErrorBody: Decodable {
    let code: String
    let message: String
    let retry_after: Int?
}

struct PreflightResponse: Decodable {
    let token: String
    let session_id: String
    let risk_score: Int
    let expires_at: Int
    let pow_challenge: PowChallengeData?
    let requires_pow: Bool?
    let requires_captcha: Bool?
}

struct PowChallengeData: Decodable {
    let nonce: String
    let difficulty: Int
    let challenge_id: String
}

struct OtpSendResponse: Decodable {
    let challenge_id: String
    let expires_at: Int
    let purpose: String
}

struct OtpVerifyResponse: Decodable {
    let access_token: String
    let refresh_token: String?
    let token_type: String
    let expires_in: Int
}

struct SessionMeResponse: Decodable {
    let phone: String
    let device_id: String
    let channel: String
    let issued_at: Int
    let expires_at: Int
}

struct ClientSignals: Encodable {
    let automation_signals: AutomationSignals?
    let timing: TimingData?

    struct AutomationSignals: Encodable {
        let jailbroken: Bool
        let debugger_attached: Bool
        let suspicious_dylibs: Bool
    }

    struct TimingData: Encodable {
        let page_load_ts: Int64
        let first_interaction_ts: Int64?
        let form_submit_ts: Int64
    }
}

struct PreflightRequest: Encodable {
    let channel: String
    let device_id: String
    let fingerprint: [String: String]?
    let client_signals: ClientSignals?
}

struct OtpSendRequest: Encodable {
    let phone: String
    let purpose: String
    let pow_solution: PowSolutionRequest?
    let captcha_token: String?
}

struct PowSolutionRequest: Encodable {
    let nonce: String
    let solution: String
    let challenge_id: String
}

struct OtpVerifyRequest: Encodable {
    let phone: String
    let code: String
    let challenge_id: String
    let device_id: String
}

File: android/app/src/main/java/com/otppoc/android/models/AuthModels.kt
package com.otppoc.android.models

data class ApiResponse<T>(
    val success: Boolean,
    val data: T?,
    val error: ApiErrorBody?
)

data class ApiErrorBody(
    val code: String,
    val message: String,
    val retry_after: Int?
)

data class PreflightResponse(
    val token: String,
    val session_id: String,
    val risk_score: Int,
    val expires_at: Long,
    val pow_challenge: PowChallengeData?,
    val requires_pow: Boolean?,
    val requires_captcha: Boolean?
)

data class PowChallengeData(
    val nonce: String,
    val difficulty: Int,
    val challenge_id: String
)

data class OtpSendResponse(
    val challenge_id: String,
    val expires_at: Long,
    val purpose: String
)

data class OtpVerifyResponse(
    val access_token: String,
    val refresh_token: String?,
    val token_type: String,
    val expires_in: Int
)

data class SessionMeResponse(
    val phone: String,
    val device_id: String,
    val channel: String,
    val issued_at: Long,
    val expires_at: Long
)

data class ClientSignals(
    val honeypot_name: String = "",
    val honeypot_email: String = "",
    val honeypot_url: String = "",
    val automation_signals: AutomationSignals?,
    val timing: TimingData?
)

data class AutomationSignals(
    // Backend-recognized fields (mobile path in risk.service.ts)
    val jailbroken: Boolean,          // maps to rooted on Android
    val debugger_attached: Boolean,
    val suspicious_dylibs: Boolean,   // maps to suspicious_libs on Android
    // Extended Android-specific signals (informational for risk scoring)
    val emulator: Boolean,
    val ui_automation: Boolean,
    val hooking_framework: Boolean,
    val screen_reader_abuse: Boolean,
    val overlay_detected: Boolean
)

data class TimingData(
    val page_load_ts: Long,
    val first_interaction_ts: Long?,
    val form_submit_ts: Long
)

data class PreflightRequest(
    val channel: String,
    val device_id: String,
    val fingerprint: Map<String, String>?,
    val client_signals: ClientSignals?
)

data class OtpSendRequest(
    val phone: String,
    val purpose: String,
    val pow_solution: PowSolutionRequest?,
    val captcha_token: String?
)

data class PowSolutionRequest(
    val nonce: String,
    val solution: String,
    val challenge_id: String
)

data class OtpVerifyRequest(
    val phone: String,
    val code: String,
    val challenge_id: String,
    val device_id: String
)
